racker/salus-event-engine-processor
# Handle Task/Monitor Change Events

# What

Adds consumers for `TaskChangeEvent`s and `MonitorChangeEvent`s.

* `TaskChangeEvent`s cause (re/un)deploys of task queries to esper
* `MonitorChangeEvent`s cause monitor property caches to be cleared

# How

1. Updated `CachedRepositoryRequests.java` to have some new `CacheEvict` methods
1. The "normal" kafka consumer properties in `application.yml` are used for these new consumers
    * The metrics consumer has been updated to provide its own custom properties
1. The partition tracking logic for the metrics consumer has been moved to its own class `PartitionTracker.java`
   * This enables multiple classes to view what partitions the consumer is responsible for
1. Moved some test logic to `MultipleConsumerTestConfig.java` since it started getting cluttered due to needing 3 instances of multiple classes

## How to test

Run all tests.

I will begin manual testing tomorrow to verify that all recent pieces of event-processor / mgmt are working and to try and identify any missing parts.

# Why

Tasks are currently only loaded at startup.  This enables it to keep track of new changes after it is already running.

Caching of monitor data was added to limit the number of db lookups.  This now clears those caches if the monitor has been modified.
